shell/supervisor.go

```diff
@@ -62,7 +63,13 @@ func (w *wrapper) Wait() (err error) {
 	done := make(chan error, 1)
 
 	go func() {
-		done <- cmd.Wait()
+		err := cmd.Wait()
+		if atomic.LoadUint32(&w.killedByCancel) > 0 {
```

Comment:
This is the only time we read it.  Why not a boolean?

Reply:
There is no such thing as an atomic boolean. There exists a package called atomic bool, but it's implemented with an atomic int32. It’s also what `sync.Once` uses.